.github/workflows/release.yaml

```
@@ -0,0 +1,193 @@
+            NEW_VERSION_SUFFIX=$(./semver get prerel ${{ env.RELEASE_TAG }})
+            [ -n "$NEW_VERSION_SUFFIX" ] && NEW_VERSION_SUFFIX="-$NEW_VERSION_SUFFIX"
+          fi
+          sed -i "s/VERSION_SUFFIX=.*/VERSION_SUFFIX=$NEW_VERSION_SUFFIX/" VERSION
+      - name: Commit with requested version and push
+        run: |
+          git fsck
+          git gc
+          git add VERSION
+          git config --global user.email ${{ github.actor }}@scality.com
+          git config --global user.name ${{ github.actor }}
+          git commit -m "Release ${{ env.RELEASE_TAG }}"
+          git push --set-upstream origin ${{ env.RELEASE_BRANCH }}
+      - name: Create and push `${{ env.RELEASE_TAG }}` tag
+        run: |
+          BODY=$(mktemp)
+
+          PATCH_MSG=""
+          if [ "${{ env.PATCH_VERSION }}" != "0" ]; then
+            PATCH_MSG="MetalK8s ${{ env.RELEASE_TAG }} is a patch release (see [the main release ${{ env.BASE_VERSION }}](https://github.com/scality/metalk8s/releases/${{ env.BASE_VERSION }}))."
+          fi
+
+          cat > $BODY <<- EOM
+          MetalK8s ${{ env.RELEASE_TAG }}
+          ===
+
+          $PATCH_MSG
+
+          Useful links
+          ---
+
+          - [Documentation](https://metal-k8s.readthedocs.io/en/${{ env.RELEASE_TAG }})
+          - [Upgrade notes](https://metal-k8s.readthedocs.io/en/${{ env.RELEASE_TAG }}/operation/upgrade.html)
+          - [Changelog](https://github.com/scality/metalk8s/blob/${{ env.RELEASE_TAG }}/CHANGELOG.md)
+
+          EOM
+
+          if [ '${{ inputs.what-is-new }}' != '' ]; then
+            echo "What's new" >> $BODY
+            echo "===" >> $BODY
+            echo "" >> $BODY
+            while IFS= read -r line; do
+              echo "- $line" >> $BODY
+            done <<< $(echo '${{ inputs.what-is-new }}' | jq -r '.[]')
+            echo "" >> $BODY
+          fi
+
+          git tag -a "${{ env.RELEASE_TAG }}" -F $BODY
+          git push origin ${{ env.RELEASE_TAG }}
+      - name: Prepare branch to merge ${{ env.RELEASE_TAG }} tag
+        run: |
+          git checkout development/${{ env.SHORT_VERSION }}
+          git pull
+          git checkout -b release/merge/${{ env.RELEASE_TAG }}
+          git merge ${{ env.RELEASE_TAG }} --no-ff
+
+          source VERSION
+
+          if [ "${{ inputs.version-type }}" == "GA" ]; then
+            VERSION_PATCH=$((VERSION_PATCH + 1))
+            sed -i "s/## Release ${{ env.BASE_VERSION }} (in development)/## Release $VERSION_MAJOR.$VERSION_MINOR.$VERSION_PATCH (in development)\n\n## Release ${{ env.BASE_VERSION }}/" CHANGELOG.md
+            git add CHANGELOG.md
+
+            sed -i "s/VERSION_PATCH=.*/VERSION_PATCH=$VERSION_PATCH/" VERSION
+          fi
+
+          VERSION_SUFFIX="-dev"
+          sed -i "s/VERSION_SUFFIX=.*/VERSION_SUFFIX=$VERSION_SUFFIX/" VERSION
+
+          git add VERSION
+          git commit -m "Bump version to $VERSION_MAJOR.$VERSION_MINOR.$VERSION_PATCH$VERSION_SUFFIX"
```

Personnally I prefer to put it in the merge commit directly 

```suggestion
          git commit --amend --no-edit
```

---

Done!